package/MDAnalysis/analysis/density.py

```diff
@@ -394,8 +402,18 @@ def _prepare(self):
                 logger.warning(msg)
             # Generate a copy of smin/smax from coords to later check if the
             # defined box might be too small for the selection
-            smin = np.min(coord, axis=0)
-            smax = np.max(coord, axis=0)
+            try:
+                smin = np.min(coord, axis=0)
+                smax = np.max(coord, axis=0)
+            except ValueError as err:
+                msg = ("No atoms in AtomGroup at input time frame. "
+                       "This may be intended; please ensure that "
+                       "your grid selection covers the atomic "
+                       "positions you wish to capture.")
+                warnings.warn(msg)
+                logger.warning(msg)
+                smin = self._gridcenter
+                smax = self._gridcenter
```

Comment:
Why are `smin` and `smax` assigned the grid center coordinates?

Please explain the logic here. What consequences does it have downstream? Does the code fail further downstream if the user has *not* supplied a proper grid selection?

Reply:
Oh, that was done because `_set_user_grid` wanted a preassigned range from the coordinates of `AtomGroup` in selection. But there is no `AtomGroup` in this case, so I initialize a zero range grid at the grid center and take the range from the user-defined grid values.

Reply:
Could you do
```python
# need to be set for _set_user_grid() but will be
# overriden by user-defined values
smin = smax = None
```
to make clearer what's happening? If `None` is not possible (even though that would be the best value to indicate that we are not using these values), just setting to 0 or `[0, 0, 0]` would still be cleaner.

Reply:
It will trigger this warning in set_user_grid if I do that.
`        if any(smin < umin) or any(smax > umax):`
`            msg = ("Atom selection does not fit grid --- "`
`                   "you may want to define a larger box")`
`            warnings.warn(msg)`
 `           logger.warning(msg)`

 The check is meant for regular situations to compare the atom selection with user-defined grid.

 In the case of no atoms in selection, I just want it to pass through. So setting smin and smax at the gridcenter will always pass the comparison with a user defined box.

Reply:
Document in a comment, please.